tests/protocol/test_wire.py

```diff
@@ -516,227 +515,209 @@ def test_get_ext_conf_with_uri(self, *args):
-        mock_successful_response = MockResponse(body=b"OK", status_code=200)
-        destination = os.path.join(self.tmp_dir, "tmp_file")
+                success = protocol.download_ext_handler_pkg(extension_url, target_file)
 
-        # Direct channel fails, host channel fails due to stale goal state, host channel succeeds after refresh.
-        # As a consequence, goal state should have been updated and host channel should have been set as default.
-        with patch("azurelinuxagent.common.utils.restutil._http_request",
-                   side_effect=[mock_failed_response, mock_failed_response, mock_successful_response]):
-            with patch(
-                    "azurelinuxagent.common.protocol.wire.WireClient.update_host_plugin_from_goal_state") as mock_update_host_plugin_from_goal_state:
-                with patch("azurelinuxagent.common.protocol.wire.WireClient.stream", wraps=protocol.client.stream) \
-                        as patch_direct:
-                    with patch(
-                            "azurelinuxagent.common.protocol.wire.WireProtocol._download_ext_handler_pkg_through_host",
-                            wraps=protocol._download_ext_handler_pkg_through_host) as patch_host:
-                        ret = protocol.download_ext_handler_pkg("uri", destination)
-                        self.assertEquals(ret, True)
-
-                        self.assertEquals(patch_host.call_count, 2)
-                        # The host channel calls the direct function under the covers
-                        self.assertEquals(patch_direct.call_count, 1 + patch_host.call_count)
-                        self.assertEquals(mock_update_host_plugin_from_goal_state.call_count, 1)
-
-                        self.assertEquals(HostPluginProtocol.is_default_channel(), True)
-
-    @skip_if_predicate_true(lambda: True, "Needs to be re-enabled before release 2.2.47")
-    @patch("azurelinuxagent.common.protocol.wire.WireClient.get_goal_state")
-    @patch("azurelinuxagent.common.protocol.hostplugin.HostPluginProtocol.get_artifact_request")
-    def test_download_ext_handler_pkg_should_not_change_default_channel_if_host_fails(self, mock_get_artifact_request,
-                                                                                      *args):
-        mock_get_artifact_request.return_value = "dummy_url", "dummy_header"
-        protocol = WireProtocol("foo.bar")
-        HostPluginProtocol.set_default_channel(False)
+                urls = http_request.get_tracked_urls()
+                self.assertEquals(success, True, 'The download should have succeeded')
+                self.assertEquals(len(urls), 2, "Unexpected number of HTTP requests: [{0}]".format(urls))
+                self.assertEquals(urls[0], extension_url, "The first attempt should have been over the direct channel")
+                self.assertTrue(urls[1].endswith('/extensionArtifact'), "The retry attempt should have been over the host channel")
```

Comment:
Do we need to validate the IP address to validate it went through the HGAP? I know the `/extensionArtifact` should be good enough to validate if the correct URI was hit.

Reply:
yes, that is one of the changes i want to do post-release